tests_e2e/orchestrator/lib/agent_test_suite.py

```diff
@@ -230,9 +238,14 @@ def _initialize(self, environment: Environment, variables: Dict[str, Any], lisa_
 
         if isinstance(environment.nodes[0], LocalNode):
             # We need to create a new VMSS.
-            # Use the same naming convention as LISA for the scale set name: lisa-<runbook name>-<run id>-e0-n0. Note that we hardcode the resource group
-            # id to "e0" and the scale set name to "n0" since we are creating a single scale set.
-            self._resource_group_name = f"lisa-{self._runbook_name}-{RUN_ID}-e0"
+            # Use the same naming convention as LISA for the scale set name: lisa-<runbook name>-<run id>-e<rg count>-n0
+            # Note that we hardcode the scale set name to "n0" since we are creating a single scale set.
+            # Resource group name cannot have any uppercase characters, because the publicIP cannot have uppercase
+            # characters in its domain name label.
+            AgentTestSuite._rg_count_lock.acquire()
+            self._resource_group_name = f"lisa-{self._runbook_name.lower()}-{RUN_ID}-e{AgentTestSuite._rg_count}"
+            AgentTestSuite._rg_count += 1
+            AgentTestSuite._rg_count_lock.release()
```

Comment:
it's a good practice to use acquire/release with try/finally to ensure the lock is released when there is an exception in the innner code